spec.md

```diff
@@ -1,8 +1,10 @@
 # Introduction
 
-This specification describes a mechanism for splitting a large
-byte stream into blocks based on their content, and a mechanism for
-organizing those blocks into a balanced tree.
+This specification describes a mechanism
+for splitting a byte stream into blocks of varying size
+with split boundaries based solely on the content of the input.
+It also describes a mechanism for organizing those blocks into a tree
+whose shape is likewise determined solely by the content of the input.
```

Comment:
N.b. not "balanced."

Reply:
It actually should be balanced (if the hash function is any good anyway...), though only probabilistically. And this is important for some use cases -- otherwise e.g. seek() would be slow for perkeep's fuse filesystem. So maybe we should qualify the balancing property but I don't think we should not mention it.

Reply:
I'm wondering if we should actually enforce min/max node sizes as well; since balance is important for lookup performance, a skewed tree (which could easily happen given our non-cryptographic hash function) could be a source of DoS vulnerabilities.

Reply:
Good point, and nicely analogous to S_min and S_max.